### DISCUSSION TITLE
How to take multiple arguments in one option

### DISCUSSION THREAD

**Discussion by lm-dt** [bkirwi/decline#484] [Q&A]
I'm trying to migrate an existing program to decline, and I have an option that takes two arguments like this:
`--at filename data`
Can this be achieved in decline? I've been able to get it to accept it as `filename data` using `.tupled` to make an `Opt[(String, String)]`, but this doesn't give me any sort of `Argument[(String, String)]` that I could use with `Opts.option`. This is my first time using decline, so I could be missing something obvious.

**Comment by bkirwi** [comment]
No, this can't be done in decline, sorry! Options can have at most one
argument.

I know that some other option parsers allow this (scallop, python's
optparse) but decline hews pretty close to the unix standards which don't
have anything similar. I appreciate that this can make porting existing
CLIs more annoying!

The usual pattern in unix / decline is to either use a single option with a
separator, or two different options. Either of those feel like they might
work for your case?

**Reply by lm-dt** [reply]
So it would be used like `--at filename|data`, where `|` is the separator?

**Reply by bkirwi** [reply]
Yeah pretty much! There's an example here: https://ben.kirw.in/decline/arguments.html#defining-your-own

I tend to prefer this style when `--at` will be passed more than once, like `cmd --at filename:data --at filename:data`. Otherwise, I often find two separate options more ergonomic. (You don't have to worry about what happens if the separator is part of the filename, etc.) If it's important that the user provides both filename and data when either are provided, you can do something like:

```
val fileOpt = Opts.option[Path]("at-file", "...");
val dataOpt = Opts.option[String]("at-data", "...");
let bothOpt = (fileOpt, dataOpt).tupled.orNone
```

Which will make sure the relationship is expressed in the usage text and error messages.
